sources/packages/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -56,6 +57,57 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
     return this.repo.save(saveLocation);
   }
 
+  /**
+   * Updating institution location.
+   * @param data Payload of updated data.
+   * @param locationId Location Id to update.
+   * @returns Updated location as InstitutionLocation.
+   */
+  async updateLocationPrimaryContact(
+    data: InstitutionLocationPrimaryContactModel,
+    locationId: number,
+  ): Promise<InstitutionLocation> {
+    const saveLocation = {
+      primaryContact: {
+        firstName: data.primaryContactFirstName,
+        lastName: data.primaryContactLastName,
+        email: data.primaryContactEmail,
+        phone: data.primaryContactPhone,
+      },
+      id: locationId,
+    };
+
+    return this.repo.save(saveLocation);
+  }
+
+  /**
+   * Updating institution location.
+   * @param data Payload of updated data.
```

Comment:
may you can combine these two function

Reply:
Even though the methods looks similar, they update 2 different set of datas in the model, so had it split.